guides/hiring/pairing.md

```diff
@@ -0,0 +1,45 @@
+# Interview Pair Programming Guidance
+
+We want to achieve the following with this guidance:
+
+- Put the candidate at ease
+- Have some consistency around the process
+- Make sure the assessor knows what to do (putting them at ease too)
+
+## Guidance
+
+1. Ask if they’ve done any pair programming before
+2. Ask if they’re familiar with any styles of pair programming. e.g. Ping-Pong, Driver-Navigator.
+3. Agree on a pairing style by explaining the rules.
+4. Set the expectation that you will be pairing with them, as an active participant.
+5. Ask if they are familiar with code katas.
+6. If they are familiar, ask them about their favourites. Make a judgement call on the complexity (e.g. katas with a grid can be too much for 30 mins)
+  - it shouldn't matter if you've tackled the specific kata before or not.
+7. If they are not familiar, or they don’t have a favourite, take them to learn.madetech.com and discuss the options.
+  - tennis or roman numerals are good for the time available.
+  - don't do refactoring katas because they solely focus on refactoring skills. 
+9. Set the expectation that the kata doesn’t need to be finished.
+10. The candidate or you can write the first test, 
+  - See how far they can get with getting a test runner setup in the language of their choice. 
+  - Help them if they find this challenging, if you can.
+11. While pairing, pause on occasion to see if they can figure stuff out on their own (especially if you have seen the kata before).
+12. If they make a decision that has many other alternative paths forward, question it to see if they have a good justification.
+13. Afterwards, thank them for pairing and shake their hand, if culturally appropriate. This goes a long way to project our culture of respect, trust and knowledge sharing. 
+
+## What we are looking for
+
+**Knowledge & Skill**
+
+- Ability to code (tests and production code).
+- (If using their own machine), ability to create a new project and execute code.
+- Pair Programming.
+- Good approach to problem-solving (TDD or otherwise).
```

Comment:
Guidance around otherwise? Can I suggest a diagram if we're not TDDing?

Reply:
Yea if people are struggling to take a TDD approach, I wont be trying to teach them in 30 mins - instead I look for other coping mechanisms

- Pen & Paper
- Diagrams
- Writing comments with thought process
- REPL-driven development

Non-exhaustive - want to let the interviewer judge the sanity of the approach

Reply:
Dealer's choice, then :)